lib/irb/history.rb

```diff
@@ -51,14 +52,19 @@ def save_history
           raise
         end
 
+        from = 0
         if File.exist?(history_file) &&
            File.mtime(history_file) != @loaded_history_mtime
-          history = history[@loaded_history_lines..-1] if @loaded_history_lines
+          from = @loaded_history_lines
           append_history = true
         end
 
+        hist = []
+        from.upto(self.class::HISTORY.size - 1) do |index|
+          hist << self.class::HISTORY[index].split("\n").join("\\\n")
+        end
```

Comment:
This code does not look rubyish compared to the original `history = history[@loaded_history_lines..-1] if @loaded_history_lines` that does not work.
How about changing `history` to an array first? then it will work fine for `Readline::HISTORY` too.

```ruby
def save_history
  history = self.class::HISTORY.to_a
```

Reply:
Done.